### PR TITLE
perf: improve utility and overhead

### PR DESCRIPTION
**Pull request by manato**
## PR Type

- Improvement

## Description

This PR includes two following modifications:
 - Accepting run time argument to set pulse width in milliseconds
 - Class-nizing jetson_gpio that has file descriptors as members so that multiple times file opening/closing  can be avoided

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ This node does not take any inputs.
 | `phase`       | double | Desired phase of the trigger relative to ToS (Top of Second) |
 | `frame_rate`  | double | Desired frequency of the trigger in Hz                       |
 | `cpu_core_id` | int    | Desired CPU core for execution\*                             |
+| `pulse_width_ms` | int | Desired pulse width (i.e., time period that GPIO pin to be high) in millisecond |
```
Comment (drwnz) [inline README.md]: Suggest: "Desired pulse width (the time period in milliseconds for the GPIO pin to remain high)"

**Review by drwnz — COMMENTED** [review]
Looks good! Just a minor suggestion forthe README.

File: README.md (hunk 1 above)
Comment (manato) [reply README.md]: @drwnz 
Thank you for your suggestion! I've applied your suggestion at c15de195b648737c8efc9e66483a18b3da0b8122

**Comment by drwnz** [comment]
LGTM!

**Pull request merged**